pkg/builtin/unary/acos.go

```diff
@@ -0,0 +1,283 @@
+				origVecCol := origVec.Col.([]int32)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				acosResult := acos.AcosInt32(origVecCol, results)
+				if nulls.Any(acosResult.Nsp) {
+					if !nulls.Any(origVec.Nsp) {
+						resultVector.Nsp = acosResult.Nsp
+					} else {
+						resultVector.Nsp.Or(acosResult.Nsp)
+					}
+				}
+				return resultVector, err
+			},
+		},
+		{ // T_int64
+			Typ:        types.T_int64,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]int64)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				acosResult := acos.AcosInt64(origVecCol, results)
+				if nulls.Any(acosResult.Nsp) {
+					if !nulls.Any(origVec.Nsp) {
+						resultVector.Nsp = acosResult.Nsp
+					} else {
+						resultVector.Nsp.Or(acosResult.Nsp)
+					}
+				}
+				return resultVector, err
+			},
+		},
+		{ // T_float32
+			Typ:        types.T_float32,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]float32)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				acosResult := acos.AcosFloat32(origVecCol, results)
+				if nulls.Any(acosResult.Nsp) {
+					if !nulls.Any(origVec.Nsp) {
+						resultVector.Nsp = acosResult.Nsp
+					} else {
+						resultVector.Nsp.Or(acosResult.Nsp)
+					}
+				}
+				return resultVector, err
+			},
+		},
+		{ // T_float64
+			Typ:        types.T_float64,
+			ReturnType: types.T_float64,
+			Fn: func(origVec *vector.Vector, proc *process.Process, _ bool) (*vector.Vector, error) {
+				origVecCol := origVec.Col.([]float64)
+				resultVector, err := process.Get(proc, 8*int64(len(origVecCol)), types.Type{Oid: types.T_float64, Size: 8})
+				if err != nil {
+					return nil, err
+				}
+				results := encoding.DecodeFloat64Slice(resultVector.Data)
+				results = results[:len(origVecCol)]
+				resultVector.Col = results
+				nulls.Set(resultVector.Nsp, origVec.Nsp)
+				acosResult := acos.AcosFloat64(origVecCol, results)
```

Comment:
perhaps we need to set the reslutVctor's col to the return value we get from the acosResult.Result

Reply:
Done